File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> check(4, 0);
    int a[] = {1, 2, 3, 4, 5};// 5 
    int b[] = {2, 1, 2, 3, 2, 4, 2, 5};// 8
    int c[] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};// 10
    
    for(int i = 0; i < answers.size(); i++){
        if(a[i % 5] == answers[i])check[1]++;
        if(b[i % 8] == answers[i])check[2]++;
        if(c[i % 10] == answers[i])check[3]++;
    }
    int mx = *max_element(check.begin(), check.end());
    for(int i = 0; i < 4; i++){
        if(mx == check[i]){
            answer.push_back(i);
        }
    }
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] a = {1, 2, 3, 4, 5};//5
        int[] b = {2, 1, 2, 3, 2, 4, 2, 5};//8
        int[] c = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};//10
        int aa = 0, bb = 0, cc = 0;
        for (int i = 0; i < answers.length; i++) {
            if (a[(i) % 5] == answers[i]) aa++;
            if (b[(i) % 8] == answers[i]) bb++;
            if (c[(i) % 10] == answers[i]) cc++;
        }
        int mx = Math.max(aa, Math.max(bb, cc));
        ArrayList<Integer> list = new ArrayList<>();
        if (aa == mx) list.add(1);
        if (bb == mx) list.add(2);
        if (cc == mx) list.add(3);
        answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}